spark-cassandra-connector-embedded/src/main/scala/com/datastax/spark/connector/embedded/EmbeddedCassandra.scala

```diff
@@ -24,16 +25,34 @@ trait EmbeddedCassandra {
       if (currentConfigTemplate != configTemplate || forceReload) {
         clearCache()
         cassandra.map(_.destroy())
-        cassandra = Some(new CassandraRunner(configTemplate))
+        cassandra = Some(new CassandraRunner(configTemplate, DefaultHost, DefaultNativePort))
         currentConfigTemplate = configTemplate
       }
     }
   }
 }
 
+/** A utility trait for integration testing.
+  * It is only used for Spark SQL cross cluster testing as second cluster.*/
+trait SecondEmbeddedCassandra {
+
+  def cassandraHost2 = SecondEmbeddedCassandra.cassandraHost
+
+  /** Switches the Cassandra server to use the new configuration if the requested configuration is different
+    * than the currently used configuration. When the configuration is switched, all the state (including data) of
+    * the previously running cassandra cluster is lost.
+    * @param configTemplate name of the cassandra.yaml template resource */
+  def useCassandraConfig2(configTemplate: String) {
+    import SecondEmbeddedCassandra._
+    cassandra.map(_.destroy())
+    cassandra = Some(new CassandraRunner(configTemplate, DefaultHost, DefaultNativePort))
+  }
```

Comment:
no cache for this one, I will refactor the test infrastructure in another ticket.